hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java

```diff
@@ -181,7 +181,11 @@
   /** Configuration key for the list of master host:ports **/
   public static final String MASTER_ADDRS_KEY = "hbase.masters";
 
-  public static final String MASTER_ADDRS_DEFAULT =  "localhost:" + DEFAULT_MASTER_PORT;
```

Comment:
Head's up, this file is `InterfaceAudience.Public`, which means removals have to go through a full release deprecation cycle.

From the sub-section titled "Client API compatibility" in [the book](http://hbase.apache.org/book.html#hbase.versioning.post10):

> An API needs to be deprecated for a whole major version before we will change/remove it.
> * An example: An API was deprecated in 2.0.1 and will be marked for deletion in 4.0.0. On the other hand, an API deprecated in 2.0.0 can be removed in 3.0.0.

You'll need to retain the hold field -- annotate it as deprecated and add a java doc explaining that the other method could be used in its place. You can file a follow-on ticket with a fixVersion of 4.0.0 as a task to delete the field once we get there (and maybe leave a TODO referencing that ticket).

Sadly this constant is a field not a method, which means we have no way to logging a WARN for clients who make use of it.

Reply:
Ah okay. Let's avoid adding any fields to `HConstants` on the feature branch if we can.

Reply:
Ack.